FirebaseDynamicLinks/Sources/FIRDynamicLinks.m

```diff
@@ -509,7 +509,20 @@ - (BOOL)handleIncomingCustomSchemeDeepLink:(NSURL *)url {
 }
 
 - (void)passRetrievedDynamicLinkToApplication:(NSURL *)url {
-  [[UIApplication sharedApplication] openURL:url options:@{} completionHandler:nil];
+  id<UIApplicationDelegate> applicationDelegate = [UIApplication sharedApplication].delegate;
+  if (applicationDelegate &&
+      [applicationDelegate respondsToSelector:@selector(application:openURL:options:)]) {
+    // pass url directly to application delegate to avoid hop into
+    // iOS handling of the universal links
+    if (@available(iOS 9.0, *)) {
```

Comment:
This if should be deleted since < 9.0 is no longer supported.

Reply:
Removed the check